xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -677,6 +686,11 @@ func (ccw *ccWrapper) NewSubConn(addrs []resolver.Address, opts balancer.NewSubC
 func (ccw *ccWrapper) UpdateAddresses(sc balancer.SubConn, addrs []resolver.Address) {
 	newAddrs := make([]resolver.Address, len(addrs))
 	for i, addr := range addrs {
+		// First merge the resolver attributes if any and the add the handshake
+		// attributes.
+		if ccw.resolverAttr != nil {
+			addr.Attributes = addr.Attributes.Merge(ccw.resolverAttr)
+		}
```

Comment:
So we add it to both the addresses we send down to the child policies AND the addresses we get from the children when subchannels are created for them?  Is that necessary?